src/v/kafka/server/snc_quota_manager.h

```diff
@@ -68,9 +68,8 @@ class snc_quota_manager
 private:
     // configuration
     config::binding<std::chrono::milliseconds> _max_kafka_throttle_delay;
-    config::binding<std::optional<uint64_t>>
-      _kafka_throughput_limit_node_in_bps;
-    config::binding<std::optional<uint64_t>>
+    config::binding<std::optional<quota_t>> _kafka_throughput_limit_node_in_bps;
+    config::binding<std::optional<quota_t>>
```

Comment:
My idea was that it would ensure that `quota_t` is the same type as the one of configuration property (otherwise compilation will fail), but without the need to mention `quota_t` in configuration.h.

Reply:
>My idea was that it would ensure that quota_t is the same type as the one of configuration property

for that you'd need a strong type.

Reply:
actually not: simply changing the type to unsigned gives
```
error: no viable conversion from 'binding<optional<long>>' to 'binding<optional<unsigned long>>'
```

Reply:
that's not a strong type, tho--you can change your typedef to match. binding doesn't hold a quota.